add support for ES6 Set and Map objects

eduardbcom opened the pull request:
Have added conditions to fix https://github.com/jashkenas/underscore/issues/2627 issues.

I've added additional functions to check if we work with Set or Map.
Let me know if it's okay or not.
I decided not to use _.isSet(obj) cuz, well, that's a problem to use underscore functions within underscore code. Sometimes I tried to call the '_.function_name_here' but it was not defined yet.

That's more like 30% percent work, so I need to have some feedback from you guys @captbaritone @akre54 

Thanks!

> let _ = require('./underscore');
undefined
> let s = new Set([1, 2, 1])
undefined
> let m = new Map([['one', 1], ['two', 2], ['one', 1.0]])
undefined
> _.each(s, function() { console.log(arguments); }) 
{ '0': 1, '1': 1, '2': Set { 1, 2 } }
{ '0': 2, '1': 2, '2': Set { 1, 2 } }
Set { 1, 2 }
> _.each(s, (key, value, s) => console.log(key, value, s))
1 1 Set { 1, 2 }
2 2 Set { 1, 2 }
Set { 1, 2 }
> _.each(m, (key, value, m) => console.log(key, value, m))
1 'one' Map { 'one' => 1, 'two' => 2 }
2 'two' Map { 'one' => 1, 'two' => 2 }
Map { 'one' => 1, 'two' => 2 }
> _.each(m, (value, key, m) => console.log(value, key, m))
1 'one' Map { 'one' => 1, 'two' => 2 }
2 'two' Map { 'one' => 1, 'two' => 2 }
Map { 'one' => 1, 'two' => 2 }
> _.isEmpty(s)
false
> _.isEmpty(m)
false
> _.size(s)
2
> _.size(m)
2
>

akre54 commented:
> I decided not to use .isSet(obj) cuz, well, that's a problem to use underscore functions within underscore code

Why do you say that? I'd rather we dogfood a function than creating a new one.

eduardbcom commented:
I've removed logic for the _.each method, let's focus on _.isEmpty and _.size for now.
You can see that I have added babel polyfill in order to work with Set and Map within karmajs.

Is it expected code for isEmpty and size ? @akre54 @captbaritone 

I am wondering why I still receive build failed, my local version (npm run test-browser and test-node) works fine.

Thx.
